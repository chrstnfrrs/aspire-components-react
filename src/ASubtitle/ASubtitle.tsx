import * as React from 'react';

import * as HeadingSelectors from '../typography/heading-selectors';
import '../typography/heading.scss';

const ASubtitle: React.FC<HeadingSelectors.IHeading> = ({
  as,
  'data-testid': testid,
  type,
  ...props
}) => {
  const componentProps = {
    ...props,
    as: as || 'subtitle-1',
    'data-testid': testid,
    type: type || 'strong',
  };

  const { Component, className } = HeadingSelectors.getElement(componentProps);

  return (
    <Component className={className} data-testid={testid}>
      {props.children}
    </Component>
  );
};

export { ASubtitle };
