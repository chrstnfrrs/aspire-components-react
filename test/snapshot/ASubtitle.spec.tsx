import { ASubtitle } from '../../src/ASubtitle/ASubtitle';
import { AS_HEADING_ENUM } from '../enums/as-heading-enum';
import { FONTSIZE_ENUM } from '../enums/fontsize-enum';
import { TYPE_HEADING_ENUM } from '../enums/type-heading-enum';

import { testComponent } from './utils/test-component';
import { testTypography } from './utils/test-typography';

describe('Given ASubtitle component is used', () => {
  const ASubtitleProps = {
    factory: {
      fontSize: FONTSIZE_ENUM,
    },
  };

  testComponent(ASubtitle, 'ASubtitle', ASubtitleProps);
  testTypography(ASubtitle, 'ASubtitle', 'as', AS_HEADING_ENUM);
  testTypography(ASubtitle, 'ASubtitle', 'type', TYPE_HEADING_ENUM);
});
