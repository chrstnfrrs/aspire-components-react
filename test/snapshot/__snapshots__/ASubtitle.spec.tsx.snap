// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given ASubtitle component is used When a developer uses ASubtitle with a className Then the the class should be present 1`] = `
<strong
  className="asdf a-subtitle-1"
>
  ASubtitle
</strong>
`;

exports[`Given ASubtitle component is used When a developer uses ASubtitle with no props Then the only class should be a-container 1`] = `
<strong
  className="a-subtitle-1"
>
  ASubtitle
</strong>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 2xl Then the class fontSize-2xl should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-2xl"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 3xl Then the class fontSize-3xl should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-3xl"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 4xl Then the class fontSize-4xl should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-4xl"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 5xl Then the class fontSize-5xl should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-5xl"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 6xl Then the class fontSize-6xl should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-6xl"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 7xl Then the class fontSize-7xl should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-7xl"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 8xl Then the class fontSize-8xl should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-8xl"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in 9xl Then the class fontSize-9xl should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-9xl"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in an invalid value Then no fontSize class should be used 1`] = `
<strong
  className="a-subtitle-1"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in base Then the class fontSize-base should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-base"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in lg Then the class fontSize-lg should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-lg"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in sm Then the class fontSize-sm should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-sm"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in undefined Then no fontSize class should be used 1`] = `
<strong
  className="a-subtitle-1"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in xl Then the class fontSize-xl should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-xl"
/>
`;

exports[`Given ASubtitle component is used When a developer uses a factory prop type When a developer uses the fontSize prop When a developer passes in xs Then the class fontSize-xs should be used 1`] = `
<strong
  className="a-subtitle-1 fontSize-xs"
/>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with as set to "h1" Then the component should render as expected 1`] = `
<strong
  className="a-h1"
>
  ASubtitle
</strong>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with as set to "h2" Then the component should render as expected 1`] = `
<strong
  className="a-h2"
>
  ASubtitle
</strong>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with as set to "h3" Then the component should render as expected 1`] = `
<strong
  className="a-h3"
>
  ASubtitle
</strong>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with as set to "h4" Then the component should render as expected 1`] = `
<strong
  className="a-h4"
>
  ASubtitle
</strong>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with as set to "h5" Then the component should render as expected 1`] = `
<strong
  className="a-h5"
>
  ASubtitle
</strong>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with as set to "h6" Then the component should render as expected 1`] = `
<strong
  className="a-h6"
>
  ASubtitle
</strong>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with as set to "subtitle-1" Then the component should render as expected 1`] = `
<strong
  className="a-subtitle-1"
>
  ASubtitle
</strong>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with as set to "subtitle-2" Then the component should render as expected 1`] = `
<strong
  className="a-subtitle-2"
>
  ASubtitle
</strong>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with type set to "h1" Then the component should render as expected 1`] = `
<h1
  className="a-subtitle-1"
>
  ASubtitle
</h1>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with type set to "h2" Then the component should render as expected 1`] = `
<h2
  className="a-subtitle-1"
>
  ASubtitle
</h2>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with type set to "h3" Then the component should render as expected 1`] = `
<h3
  className="a-subtitle-1"
>
  ASubtitle
</h3>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with type set to "h4" Then the component should render as expected 1`] = `
<h4
  className="a-subtitle-1"
>
  ASubtitle
</h4>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with type set to "h5" Then the component should render as expected 1`] = `
<h5
  className="a-subtitle-1"
>
  ASubtitle
</h5>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with type set to "h6" Then the component should render as expected 1`] = `
<h6
  className="a-subtitle-1"
>
  ASubtitle
</h6>
`;

exports[`Given ASubtitle component is used When rendering ASubtitle with type set to "strong" Then the component should render as expected 1`] = `
<strong
  className="a-subtitle-1"
>
  ASubtitle
</strong>
`;
